Sources/Subs-Db-postgresql.php

```diff
@@ -202,6 +202,13 @@ function smf_db_replacement__callback($matches)
 				smf_db_error_backtrace('Wrong value type sent to the database. Date expected. (' . $matches[2] . ')', '', E_USER_ERROR, __FILE__, __LINE__);
 		break;
 
+		case 'time':
+			if (preg_match('~^([0-1]?\d|2[0-3]):([0-5]\d):([0-5]\d)$~', $replacement, $time_matches) === 1)
+				return sprintf('\'%02d:%02d:%02d\'', $time_matches[1], $time_matches[2], $time_matches[3]);
```

Comment:
> Is there any place in smf where the time format is defined?

This if statement is that definition. The regular expression `^([0-1]?\d|2[0-3]):([0-5]\d):([0-5]\d)$` will only match valid times given in 24 hour format with both minutes and seconds defined. If the input value matches, the sprintf function rewrites it to ensure that it is in exactly the correct format. The return value is therefore absolutely guaranteed to be a time literal in HH24:MM:SS format.

Input|Result
-----|------
19:45:32| '19:45:32'
03:30:00|'03:30:00'
3:30:00|'03:30:00'
23:15|error
23:15:00|'23:15:00'
10:30 AM|error
32:30:00|error
13:68:21|error
13:22:99|error
2016-08-03|error
foostring|error

As you can see, the regular expression is willing to tolerate input where the hour leaves off the leading zero, but the sprintf function will always add the missing zero if it is not there.